jcommune-view/jcommune-web-view/src/test/java/org/jtalks/jcommune/test/AdministrationControllerTest.groovy

```diff
@@ -109,12 +107,117 @@ class AdministrationControllerTest extends Specification {
+            dto.setId(savedGroupId)
+            dto.setName(alphanumeric(GROUP_NAME_MAX_LENGTH))
+            dto.setDescription(alphanumeric(GROUP_DESCRIPTION_MAX_LENGTH))
+            groups.edit(dto, session)
+        then:
+            groups.isExist(dto.name)
+        and: 'All fields of edited group (except ID) has been updated'
+            def edited = groupsService.getById(savedGroupId)
+            savedGroupId == edited.id
+            dto.name == edited.name
+            dto.description == edited.description
+    }
+
+    def 'must not be able to edit not existing group'() {
+        given: 'Admin signed in and forum has some existing group'
+            def session = users.signInAsAdmin(forum)
+            def savedGroupId = groupsService.save(Groups.random())
+        when: 'Admin attempts to edit not existing group'
+            def dto = new GroupAdministrationDto()
+            dto.setId(savedGroupId + 1)
+            dto.setName(alphanumeric(GROUP_NAME_MAX_LENGTH))
+            dto.setDescription(alphanumeric(GROUP_DESCRIPTION_MAX_LENGTH))
+            groups.edit(dto, session)
+        then:
+            thrown(WrongResponseException)
+    }
+
+    def 'after changing group name must not be able to find group by old name'() {
+        given: 'Admin signed in and forum has some existing group'
+            def session = users.signInAsAdmin(forum)
+            def savedGroupId = groupsService.save(Groups.random())
+            def savedGroupName = groupsService.getById(savedGroupId).name
+        when: 'Admin attempts to edit group'
+            def dto = new GroupAdministrationDto()
+            dto.setId(savedGroupId)
+            dto.setName(alphanumeric(GROUP_NAME_MAX_LENGTH))
+            dto.setDescription(alphanumeric(GROUP_DESCRIPTION_MAX_LENGTH))
+            groups.edit(dto, session)
+        then:
+            groups.assertDoesNotExistWithName(savedGroupName)
+    }
+
+    def 'must not be able to set invalid name or description'() {
+        given: 'Admin signed in and forum has some existing group'
+        def session = users.signInAsAdmin(forum)
+        def savedGroupId = groupsService.save(Groups.random())
+        when: 'Admin attempts to edit group'
+        def dto = new GroupAdministrationDto()
+        dto.setId(groupId.toInteger())
+        dto.setName(groupName)
+        dto.setDescription(groupDescription)
+        groups.edit(dto, session)
+        then: 'Validation error occurs'
+        def e = thrown(ValidationException)
+        [errorMessage] == e.defaultErrorMessages
+        and: 'Group is not edited'
+        groups.assertDoesNotExistWithName(dto.name)
+        where:
+        groupId                 |groupName           | groupDescription    |errorMessage                                                    | caseName
+        savedGroupId.toString() |''                  | alphanumeric(0,255) |"Group name length must be between 1 and 100 characters"        | 'Group name is empty'
```

Comment:
Validation rules should be checked in unit tests. Component tests are very slow and we want to write as few of them as possible. So:

- Unit tests check the validation rules
- Component tests check that those rules are invoked.